libsledge/src/memory_instructions.c

```diff
@@ -12,55 +12,55 @@ extern thread_local struct sledge_abi__wasm_module_instance sledge_abi__current_
 INLINE uint32_t
 instruction_memory_size()
 {
-	return (uint32_t)(sledge_abi__current_wasm_module_instance.memory.size / WASM_PAGE_SIZE);
+	return (uint32_t)(sledge_abi__current_wasm_module_instance.memory.size / (uint64_t)WASM_PAGE_SIZE);
 }
 
 // These functions are equivalent to those in wasm_memory.h, but they minimize pointer dereferencing
 INLINE float
 get_f32(uint32_t offset)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert(offset + sizeof(float) <= sledge_abi__current_wasm_module_instance.memory.size);
+	assert((uint64_t)offset + sizeof(float) <= sledge_abi__current_wasm_module_instance.memory.size);
```

Comment:
same as above. why cast here?

Reply:
Yes, it's a uint64_t. UNIT32_MAX is the largest size for wasm32, but it's one too large to fit in a uint32_t.  I've been bitten by type issues in this layer before, so I ended up making everything pretty explicit.